File: Day 5/day5.py
def organize_shoes(shoes:list[dict]) -> list[int]:
  result = []
  dict_aux = {}
  for shoe in shoes:
    tipo, size = shoe['type'], shoe['size']
    dict_aux.setdefault(size,'')

    reverse_par = 'R' if tipo == 'I' else 'I'
    if reverse_par in dict_aux[size]:
      dict_aux[size] = dict_aux[size].replace(reverse_par, '')
      result.append(size)
    else:
      dict_aux[size] = dict_aux[size] + tipo
 
  return result

File: Day 5/day5.ts
function orginzeShoes(shoes: {type: string, size: number}[]): number[]{
    const result: number[] = [];
    const dictAux: Record<number, string> = {};

    for (const shoe of shoes){
        const {type, size} = shoe;

        if (!(size in dictAux)){ dictAux[size] = '';}

        const reversePar = type === 'I'? 'R' : 'I';

        if (dictAux[size].includes(reversePar)){
            dictAux[size] = dictAux[size].replace(reversePar,'')
            result.push(size)
        }
        else {dictAux[size] += type;}
    }

    return result
}